source/Core/Utils.php

```diff
@@ -997,19 +998,20 @@ public function showOfflinePage()
      * @param string $sUrl               URL to be redirected
      * @param bool   $blAddRedirectParam add "redirect" param
      * @param int    $iHeaderCode        header code, default 302
-     *
-     * @return null or exit
+     * @throws RedirectException
+     * @return no-return
```

Comment:
I understand what you mean by `no-return`, but I would suggest using `void` and express the nature of this method in the description of the `@return` annotation. What do you think?

```suggestion
     * @throws RedirectException When redirect cycle is detected
     * @return void Method does not return in case redirect can be executed
```

Reply:
"no-return" can be used by static code analysers  
and is supported by psalm, phpstan and others:
- https://psalm.dev/docs/annotating_code/type_syntax/atomic_types/
- https://phpstan.org/writing-php-code/phpdoc-types
- phpstorm
and there is open pull request to get it into php-fig

https://github.com/php-fig/fig-standards/pull/1191

I understand that it is still not standardised, but it helps to find issues when developing.

Reply:
TIL :wink: 
I checked the same page in the PHPStan docs, but for some reason I couldn't find it at my first try, now it worked. In this case I would also omit the description :wink:

Reply:
Hi guys, just wanted to let you know, that Matt and Ondrej are working on a RFC to get this into the PHP Core: https://wiki.php.net/rfc/noreturn_type

&lt;/OT&gt;